### PR TITLE
mon/scrub: log error details of store access failures

### PR DESCRIPTION
**Pull request by YiteGu**
Fixes: https://tracker.ceph.com/issues/64824

## Contribution Guidelines
- To sign and title your commits, please refer to [Submitting Patches to Ceph](https://github.com/ceph/ceph/blob/main/SubmittingPatches.rst).

- If you are submitting a fix for a stable branch (e.g. "quincy"), please refer to [Submitting Patches to Ceph - Backports](https://github.com/ceph/ceph/blob/master/SubmittingPatches-backports.rst) for the proper workflow.

- When filling out the below checklist, you may click boxes directly in the GitHub web UI.  When entering or editing the entire PR message in the GitHub web UI editor, you may also select a checklist item by adding an `x` between the brackets: `[x]`.  Spaces and capitalization matter when checking off items this way.

## Checklist
- Tracker (select at least one)
  - [x] References tracker ticket
  - [ ] Very recent bug; references commit where it was introduced
  - [ ] New feature (ticket optional)
  - [ ] Doc update (no ticket needed)
  - [ ] Code cleanup (no ticket needed)
- Component impact
  - [ ] Affects [Dashboard](https://tracker.ceph.com/projects/dashboard/issues/new), opened tracker ticket
  - [ ] Affects [Orchestrator](https://tracker.ceph.com/projects/orchestrator/issues/new), opened tracker ticket
  - [x] No impact that needs to be tracked
- Documentation (select at least one)
  - [ ] Updates relevant documentation
  - [x] No doc update is appropriate
- Tests (select at least one)
  - [ ] Includes [unit test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/tests-unit-tests/)
  - [ ] Includes [integration test(s)](https://docs.ceph.com/en/latest/dev/developer_guide/testing_integration_tests/)
  - [ ] Includes bug reproducer
  - [x] No tests

<details>
<summary>Show available Jenkins commands</summary>

- `jenkins retest this please`
- `jenkins test classic perf`
- `jenkins test crimson perf`
- `jenkins test signed`
- `jenkins test make check`
- `jenkins test make check arm64`
- `jenkins test submodules`
- `jenkins test dashboard`
- `jenkins test dashboard cephadm`
- `jenkins test api`
- `jenkins test docs`
- `jenkins render docs`
- `jenkins test ceph-volume all`
- `jenkins test ceph-volume tox`
- `jenkins test windows`
- `jenkins test rook e2e`
</details>

### CODE REVIEW

**Comment by YiteGu** [comment]
@rzarzynski help review

**Comment by ronen-fr** [comment]
@aclamk, can you take a look? does treating the error val received from the store (which, if I understand correctly, is usually rocksdb) as an input to cpp_strerror() make sense? or should we just call ceph_abort with the actual int value in the message?

**Comment by YiteGu** [comment]
@ronen-fr @aclamk related tracker: https://tracker.ceph.com/issues/64824

File: src/mon/Monitor.cc [hunk 1]
```diff
@@ -5708,7 +5708,12 @@ bool Monitor::_scrub(ScrubResult *r,
 
     bufferlist bl;
     int err = store->get(k.first, k.second, bl);
```
Comment (aclamk) [inline src/mon/Monitor.cc:5710]: When store is RocksDB, the only possible err are 0 and -ENOENT.
All other errors are handled with `ceph_abort_msg(s.getState());` that prints RocksDB internal failure reason.

I guess this is valuable extension; we will see that the reason was -ENOENT.

File: src/mon/Monitor.cc [hunk 2]
```diff
@@ -5708,7 +5708,12 @@ bool Monitor::_scrub(ScrubResult *r,
...
-    ceph_assert(err == 0);
+    if (err != 0) {
+      derr << __func__ << " store got: " << cpp_strerror(err)
+                       << " prefix: " << k.first << " key: " << k.second
+                       << dendl;
+      ceph_assert(0);
```
Comment (aclamk) [inline src/mon/Monitor.cc:5715]: NIT: ceph_abort() is the same.

File: src/mon/Monitor.cc (hunk 1 above)
Comment (YiteGu) [reply src/mon/Monitor.cc:5710]: Yep, so we can still use cpp_strerror(err) here, right?

File: src/mon/Monitor.cc (hunk 2 above)
Comment (YiteGu) [reply src/mon/Monitor.cc]: done

**Comment by YiteGu** [comment]
jenkins test api

**Review by rzarzynski — APPROVED** [review]

**Comment by kamoltat** [comment]
RADOS APPROVED https://tracker.ceph.com/projects/rados/wiki/MAIN#httpstrackercephcomissues67842

**Pull request merged**